lib/internal/modules/esm/resolve.js

```diff
@@ -1015,8 +1015,6 @@ function resolveAsCommonJS(specifier, parentURL) {
 // TODO(@JakobJingleheimer): de-dupe `specifier` & `parsed`
 function checkIfDisallowedImport(specifier, parsed, parsedParentURL) {
   if (parsedParentURL) {
-    const parentURL = fileURLToPath(parsedParentURL?.href);
-
```

Comment:
There was a reason for this (I'm pretty sure not doing this will break something), but I can't remember what it was. @bmeck do you remember?

Reply:
OH! I partially remember! It's when something is wrong with `parsedParentURL`—that's why it uses `?.`. Without this, the error message was too confusing, like " is not allowed" (where there should be something at the front).

Reply:
We're checking that `parsedParentURL` is truthy the line just before that, I don't think the `?.` could ever take effect. Arguably it's fixing a bigger issue than a confusing error message, I'd be inclined to land this anyway. We can always improve things later if the original problem comes back. wdyt?